back/api/models.py

```diff
@@ -2,10 +2,50 @@
 
 # Create your models here.
 
+"""Lina s'occupe de l'authentification et des classes Student et Teacher
+ici Member peut être un Student ou un Teacher"""
+
 class Member(models.Model):
     username = models.CharField(null=True, blank=True, max_length = 255)
     updated = models.DateTimeField(auto_now=True)
     created = models.DateTimeField(auto_now_add=True)
 
-    def __str__(self):
-        return self.username[0:50]
\ No newline at end of file
+    def __str__(self) -> str:
+        return self.username[0:50]
+
+class Exercise(models.Model):
+    """ Exerices composé de l'énoncé, de la solution écrite par le professeur,
+    des test à lancer (défini par le professeur) et des output de sortie qu'il
+    doit ressortir (pas forcément défini par le professeur)"""
+    subject = models.CharField(null=True, max_length = 255)
+    solution = models.CharField(null=True, max_length = 255)
+    test_input = models.CharField(null=True, max_length = 255)
+    correct_output = models.CharField(null=True, max_length = 255)
```

Comment:
Think a better way is : 
```suggestion
    statement = models.TextField()
    solution = PythonCodeField(blank=True, null=True)
    test_input = models.CharField(null=True, max_length = 255)
    correct_output = models.TextField()
``` 

I let you check how to use `PythonCodeField` [here](https://github.com/chrisspen/django-python-code-field) because it seems interesting according to what we want to do.

Finally, I'm not convinced that a text field is the best type to store in the DB for `test_input`.. Let think about an other way.

Reply:
I have made the changes

Reply:
:100: